File: output/gemma4-e4b/avatar/Java/Python/atcoder_ABC145_B.py
import sys

def solve():
    # Read n (integer)
    try:
        # Read the first line for n
        n_line = sys.stdin.readline().strip()
        if not n_line:
            return
        n = int(n_line)
    except Exception:
        return

    # Read str (string)
    try:
        # Read the second line for the string
        str_val = sys.stdin.readline().strip()
    except Exception:
        str_val = ""

    # Calculate the first half of the string, using n as the length indicator
    # Python slicing handles this directly: str_val[:k] gets the first k characters.
    a = str_val[:n // 2]

    # Check if str_val equals the concatenation of a with itself
    if str_val == a + a:
        print("Yes")
    else:
        print("No")

solve()

File: output/gemma4-e4b/avatar/Python/Java/atcoder_ABC145_B.java
import java.util.Scanner;

public class atcoder_ABC145_B {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        // Read N
        int n = scanner.nextInt();
        
        // Read S
        String s = scanner.next();
        
        if (n % 2 != 0) {
            System.out.println("No");
        } else {
            int mid = n / 2;
            
            // Check if the first half equals the second half
            // s.substring(0, mid) is the first half (length mid)
            // s.substring(mid) is the second half (starting at mid)
            
            // Note: We assume s has at least length n, as implied by the Python logic.
            if (s.substring(0, mid).equals(s.substring(mid))) {
                System.out.println("Yes");
            } else {
                System.out.println("No");
            }
        }
        
        scanner.close();
    }
}